docs/tutorials/apiIntegration/userSponsored.md

```diff
@@ -0,0 +1,304 @@
+---
+sidebar_label: "User sponsored"
+sidebar_position: 1
+---
+
+# User sponsored
+This tutorial represents the API integration flow where smart contract wallet has the native tokens to sponsor the transaction.
+
+## Pre-requisites: 
+
+- Biconomy bundler url (refer to the [docs](/dashboard#bundler-keys) to get the same)
+- Partial user operation, where
+    - sender is the smart account Address
+    - nonce can be calculated using the smart account contract methods.
+    - initCode will be 0x, if the account is already deployed or can be fetched using contract methods.
+    - paymasterAndData will be 0x, given the smart account has native tokens to pay for the gasFees.
+    - callData is the abi encoded form of transaction.
+    - It also requires putting a semi-valid/ dummy signature (e.g. a signature in the right length).
+
+```ts
+type UserOperation = {
+  sender: string;
+  nonce: BigNumberish;
+  initCode: BytesLike;
+  callData: BytesLike;
+  callGasLimit: BigNumberish;
+  verificationGasLimit: BigNumberish;
+  preVerificationGas: BigNumberish;
+  maxFeePerGas: BigNumberish;
+  maxPriorityFeePerGas: BigNumberish;
+  paymasterAndData: BytesLike;
+  signature: BytesLike;
+}
```

Comment:
BigNumberish and BytesLike types are a little confusing, they are very specific to ethers. I would suggest keeping it some generic type as people not using JS environment and not using ethers will get confused, so would suggest to keep like this:

```
type UserOperation = {
  sender: string; // hex or number in string
  nonce: string; // hex or number in string
  initCode: string; // hex or number in string
  callData: string; // hex or number in string
  callGasLimit: string; // hex or number in string
  verificationGasLimit: string; // hex or number in string
  preVerificationGas: string; // hex or number in string
  maxFeePerGas: string; // hex or number in string
  maxPriorityFeePerGas: string; // hex or number in string
  paymasterAndData: string; // hex or number in string
  signature: string; // hex or number in string
}
```